WooCommerce/Classes/Authentication/Navigation Exceptions/WrongAccountErrorViewModel.swift

```diff
@@ -0,0 +1,138 @@
+
+    init(siteURL: String?, sessionManager: SessionManagerProtocol =  ServiceLocator.stores.sessionManager) {
+        self.siteURL = siteURL ?? Localization.yourSite
+        self.defaultAccount = sessionManager.defaultAccount
+    }
+
+    // MARK: - Data and configuration
+    var userEmail: String {
+        guard let account = defaultAccount else {
+            DDLogWarn("⚠️ Present account mismatch UI for \(siteURL) without a default account")
+
+            return ""
+        }
+        return account.email
+    }
+
+    var userName: String {
+        guard let account = defaultAccount else {
+            DDLogWarn("⚠️ Present account mismatch UI for \(siteURL) without a default account")
+
+            return ""
+        }
+
+        return account.displayName
+    }
+
+    var signedInText: String {
+        guard let account = defaultAccount else {
+            DDLogWarn("⚠️ Present account mismatch UI for \(siteURL) without a default display account")
+
+            return ""
+        }
+
+        return String.localizedStringWithFormat(Localization.signedInMessageFormat,
+                                                account.username)
+    }
+
+    let logOutTitle: String = Localization.wrongAccountMessage
+
+    let logOutButtonTitle: String = Localization.logOutButtonTitle
+
+    let image: UIImage = .errorImage
+
+    var text: NSAttributedString {
+        let font: UIFont = .body
+        let boldFont: UIFont = font.bold
+
+        let boldSiteAddress = NSAttributedString(string: siteURL.trimHTTPScheme(),
+                                                           attributes: [.font: boldFont])
+        let message = NSMutableAttributedString(string: Localization.errorMessage)
+
+        message.replaceFirstOccurrence(of: "%@", with: boldSiteAddress)
+
+        return message
+    }
+
+    let isAuxiliaryButtonHidden = false
+
+    let auxiliaryButtonTitle = Localization.findYourConnectedEmail
+
+    let primaryButtonTitle = Localization.primaryButtonTitle
+
+    // MARK: - Actions
+    func didTapPrimaryButton(in viewController: UIViewController?) {
+        guard let navigationController = viewController?.navigationController else {
+            return
+        }
+
+        let storePicker = StorePickerViewController()
+        storePicker.configuration = .listStores
+        //storePickerCoordinator?.onDismiss = onDismiss
+        navigationController.pushViewController(storePicker, animated: true)
+    }
+
+    func didTapAuxiliaryButton(in viewController: UIViewController?) {
+        let fancyAlert = FancyAlertViewController.makeNeedHelpFindingEmailAlertController()
+        fancyAlert.modalPresentationStyle = .custom
+        fancyAlert.transitioningDelegate = AppDelegate.shared.tabBarController
+        viewController?.present(fancyAlert, animated: true)
+    }
+
+    func didTapLogOutButton(in viewController: UIViewController?) {
+        // Log out and pop
+        ServiceLocator.stores.deauthenticate()
```

Comment:
Do you think it's worth it to inject these dependencies, would it help on unit tests?

Reply:
Sure. Just for the record, DI is not the only way to unit test services get hit, ServiceLocator also exposes setters precisely for tests

Reply:
True, but we have faced scenarios where a test sets something in `ServiceLocator` but then does not clean it correctly, which led to inconsistencies in other tests where is not clear that `ServiceLocator` is used behind the scenes.

This is easily solved by exposing all possible dependencies in the `init` :-)